ooiservices/tests/test_user.py

```diff
@@ -104,17 +104,31 @@ def test_create_user_route(self):
         self.assertTrue(response.status_code == 401)
 
         #Test authorized
-        response = self.client.post(url_for('main.create_user'), headers=self.get_api_headers('admin', 'test'), data=json.dumps({'email': 'test@test', 'password': 'testing', 'repeatPassword': 'testing', 'phonenum': '1234', 'username': 'test_user'}))
-        self.assertTrue(response.status_code == 201)
+        response = self.client.post(url_for('main.create_user'), headers=self.get_api_headers('admin', 'test'), data=json.dumps({'email': 'test@test', 'password': 'testing', 'repeatPassword': 'testing', 'phonenum': '1234', 'username': 'test_user', 'role_id':3}))
+        self.assertEquals(response.status_code, 201)
 
         #Test duplicate user
-        response = self.client.post(url_for('main.create_user'), headers=self.get_api_headers('admin', 'test'), data=json.dumps({'email': 'test@test', 'password': 'testing', 'repeatPassword': 'testing', 'phonenum': '1234', 'username': 'test_user'}))
+        response = self.client.post(url_for('main.create_user'), headers=self.get_api_headers('admin', 'test'), data=json.dumps({'email': 'test@test', 'password': 'testing', 'repeatPassword': 'testing', 'phonenum': '1234', 'username': 'test_user', 'role_id':3}))
         self.assertTrue(response.status_code == 409)
 
         #Test password match
-        response = self.client.post(url_for('main.create_user'), headers=self.get_api_headers('admin', 'test'), data=json.dumps({'email': 'test@test', 'password': 'testing', 'repeatPassword': 'testing2', 'phonenum': '1234', 'username': 'test_user2'}))
+        response = self.client.post(url_for('main.create_user'), headers=self.get_api_headers('admin', 'test'), data=json.dumps({'email': 'test@test', 'password': 'testing', 'repeatPassword': 'testing2', 'phonenum': '1234', 'username': 'test_user2', 'role_id':3}))
         self.assertTrue(response.status_code == 409)
 
+    def test_create_user_with_role(self):
+        UserScope.insert_scopes()
+        UserScopeLink.insert_scope_link()
+        UserRoleUserScopeLink.insert_roles_scope()
+
+        #Test authorized
+        response = self.client.post(url_for('main.create_user'), headers=self.get_api_headers('admin', 'test'), data=json.dumps({'email': 'test@notduplicate', 'password': 'testing', 'repeatPassword': 'testing', 'phonenum': '1234', 'username': 'test_user_0', 'role_id':1}))
+        self.assertTrue(response.status_code == 201)
+        data = json.loads(response.data)
+
+        user = User.query.filter(User.id == data['id']).first()
+        scopes = [s.scope.id for s in user.scopes]
+        self.assertTrue(len(scopes)>0)
```

Comment:
This will be replaced with a GET `/user/scopes` call or something when it gets implemented.